src/map/Map.js

```diff
@@ -440,10 +440,15 @@ L.Map = L.Evented.extend({
 	// @method setMinZoom(zoom: Number): this
 	// Sets the lower limit for the available zoom levels (see the [minZoom](#map-minzoom) option).
 	setMinZoom: function (zoom) {
+		var oldZoom = this.options.minZoom;
 		this.options.minZoom = zoom;
 
-		if (this._loaded && this.getZoom() < this.options.minZoom) {
-			return this.setZoom(zoom);
+		if (this._loaded && oldZoom !== zoom) {
+			if (this.getZoom() < this.options.minZoom) {
+				return this.setZoom(zoom);
+			} else {
+				this.fire('zoomlevelschange');

```

Comment:
Should this really be in the `else`, shouldn't this event always fire if `oldZoom !== zoom`?

Reply:
Thanks for review :)

Zoom controls `_updateDisabled` handler  [listens for two events](https://github.com/Leaflet/Leaflet/blob/60e1965e3340bb7af6f5e33de5b9b3ad6043b285/src/control/Control.Zoom.js#L49) - `zoomend` and `zoomlevelschange`.

`this.setZoom(zoom)` fires `zoomend` event, so controls will update properly.

Since `_updateDisabled` is quite expensive (it removes and adds classNames), `zoomlevelschange` should not be fired unnecessarily.

Reply:
Ok, yeah... but for anyone else listening to _just_ `zoomlevelschange`, this will be an issue, or am I missing something? This fix is not only to fix the layers control, but also for other code using this event, right?

Reply:
I don't think this is correct as it breaks ```zoomlevelschange``` contract.  ```zoomlevelschange``` should be fired every time the number of zoom levels change, third party plugins etc will depend on this.

If updates to the zoom control are a concern than this should be fixed within the zoom control.

Reply:
You are right, I'll fix that